test/e2e/upgrades/apparmor.go

```diff
@@ -69,7 +69,7 @@ func (t *AppArmorUpgradeTest) Setup(f *framework.Framework) {
 // pod can still consume the secret.
 func (t *AppArmorUpgradeTest) Test(f *framework.Framework, done <-chan struct{}, upgrade UpgradeType) {
 	<-done
-	if upgrade == MasterUpgrade || upgrade == ClusterUpgrade {
+	if upgrade == MasterUpgrade {
```

Comment:
Master upgrades could result in pods on the master being restarted / stopped drained (that's certainly the case on some ansible based k8s deployment scripts I've seen, and any that depend on golden images).  It's possible this runs right into the ambiguity of defining what a master is, in which case the test probably has to assume the specified pod is not running on a master, which isn't unreasonable (except for single node clusters).

Reply:
Right, I was assuming that the master would be tainted, but didn't think about the single node case. Do you have a suggestion for how to fix it (is there a reliable way to identify master nodes?). Without this check, I'm not sure this test makes a whole lot of sense (it's sort of just checking a box in either case).